src/main/scala/BintrayPlugin.scala

```diff
@@ -0,0 +1,194 @@
+package bintray
+
+import bintry.Attr
+import sbt.{ AutoPlugin, Credentials, Global, Path, Resolver, Setting, Task, Tags, ThisBuild }
+import sbt.Def.{ Initialize, setting, task, taskDyn }
+import sbt.Keys._
+import sbt.Path.richFile
+
+object BintrayPlugin extends AutoPlugin {
```

Comment:
Conceptually it's the same as plugin (delivers settings to builds and projects), but it's improved in details. New features are:
-  enabling/disabling/triggering
- auto import
- plugin dependency tracking

All of the sbt's default features like `compile` are now provided by auto plugins as well.

In sbt 0.12 plugins, one of the annoying things for new user was having to put `fooSettings` in `build.sbt` in addition to writing an entry in `project/plugins.sbt`. Plugin authors were heavily discouraged from using `def settings` because then the build users can't back out of them for multi-project builds. Auto plugin solves this by providing explicit `enablePlugins(...)` and `disablePlugins(...)` method on build. This lets us write opt-in or opt-out (by default enabled) plugins that adds settings automatically.

Another annoying thing about sbt 0.12 plugins was the `import`-everything behavior. This is now more tightly controlled. Only the terms/types under `autoImport` are automatically imported.

Another annoying thing about sbt 0.12 plugins, if you have a plugin that depends on another plugin (e.g. coffeescript depends on JsEngine, which depends on web etc.) then the build user needed to place the `fooSettings` and `barSettings` in the exact order. That ordering is now automated by each plugin declaring its dependent plugins (`def requires`).

There's a tiny bit more ceremonial code for the plugin authors, but I think the build user experience is overall better.

- https://typesafe.com/blog/preview-of-upcoming-sbt-10-features-read-about-the-new-plugins
- http://www.scala-sbt.org/0.13/docs/Plugins.html
- http://www.scala-sbt.org/0.13/docs/Plugins-Best-Practices.html